File: contents/bubble_sort/code/javascript/bubble.js
function bubbleSort(arr) {
  let tmp;
  for (let i = 0; i < arr.length; i++) {
    for (let k = 0; k < arr.length - 1; k++) {
      if (arr[k] > arr[k + 1]) {
        tmp = arr[k];
        arr[k] = arr[k + 1];
        arr[k + 1] = tmp;
      }
    }
  }
}

function main() {
  const testArray = [1, 3, 2, 4, 5, 10, 50, 7, 1.5, 0.3];
  bubbleSort(testArray);
  console.log(testArray);
}

main();

File: contents/bubble_sort/code/java/Bubble.java
public class Bubble {
    static void bubbleSort(int[] arr) {
        for (int r = arr.length - 1; r > 0; r--) {
            for (int i = 0; i < r; i++) {
                if(arr[i] > arr[i + 1]) {
                    int tmp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = tmp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] test = new int[]{20, -3, 50, 1, -6, 59};

        System.out.println("Unsorted array :");
        for (int i = 0; i < test.length; i++) {
            System.out.print(test[i] + " ");
        }

        bubbleSort(test);

        System.out.println("\n\nSorted array :");
        for (int i = 0; i < test.length; i++) {
            System.out.print(test[i] + " ");
        }
        System.out.println("");
    }
}
